src/components/CategoryFilter.js

```diff
@@ -0,0 +1,43 @@
+import React, { useEffect } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import categories from '../data/categories';
+
+const CategoryFilter = props => {
+  const { handleFilterChange, filter } = props;
+
+  useEffect(() => {
+    const select = document.querySelector('.filter');
+    if (filter === 'All') {
+      select.selectedIndex = 0;
+    } else {
+      select.selectedIndex = categories.indexOf(filter) + 1;
+    }
+  }, [filter]);
+
```

Comment:
**[Required]** Is there any significance of this piece of code?

Reply:
yes, it uses state.filter to display the selected filter option, otherwise re-render reverts to All instead of e.g. History when History is the selected option